### ISSUE TITLE
Log Softmax와 Cross Entropy,  NLL Loss

### ISSUE THREAD

**Issue by gladys59** [deep-learning-with-projects/deep-learning-with-projects#18]
안녕하세요. Part 3 MNIST 분류 코드에서 코드 에러라고 생각되어지는 부분을 발견해서 글 남겨요.

pytorch에서 cross entropy 함수는 log softmax를 포함하고 있어서(cross entropy = log softmax + nll loss) model의 마지막 layer가 선형 layer로 끝나야 한다고 알고 있는데, 모델의 마지막 layer에서 log softmax가 씌워지는데 loss function으로 cross entropy가 사용되어 있더라구요.
이렇게 되면 log softmax가 이중으로 사용되게 되는데,
model의 마지막 layer에 log softmax를 씌우는 부분을 없애거나, loss function으로 nll_loss를 사용해야 하지 않나 싶습니다. 

https://stackoverflow.com/questions/65192475/pytorch-logsoftmax-vs-softmax-for-crossentropyloss
https://velog.io/@skyhelper/CrossEntorpyLoss-NLLLoss-%EB%AC%B4%EC%97%87%EC%9D%B4-%EB%8B%A4%EB%A5%B8%EA%B0%80

참고하셔서 검토해주시면 감사하겠습니다.
